channeldb/kvdb/etcd/commit_queue.go

```diff
@@ -0,0 +1,139 @@
+// +build kvdb_etcd
+
+package etcd
+
+import (
+	"context"
+	"sync"
+)
+
+// commitQueueSize is the maximum number of commits we let to queue up. All
+// remaining commits will block on commitQueue.Add().
+const commitQueueSize = 100
+
+// commitQueue is a simple execution queue to manage conflicts for transactions
```

Comment:
Very cool idea! So I take it the assumption here is that we can increase overall throughput by catching conflicts early (locally) vs remotely, as that entails the extra machinery on the DB end, as well as the extra round trip time and bandwidth due to the retires?

Reply:
Exactly, we spare a lot of extra round trips by queuing up conflicting transactions which in turn also increases overall throughput.